sdk/samples/http_client/curl/src/azure_core_with_curl.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @brief Simulates customer application that is linked with azure-core and azure-transport-curl
+ *
+ */
+
+#include <http/http.hpp>
+
+#include <iostream>
+
+using namespace azure::core;
+using namespace std;
+
+int main()
+{
+  string host("https://httpbin.org/get");
+  cout << "testing curl from transport" << endl << "Host: " << host << endl;
+
+  auto request = http::Request(http::HttpMethod::GET, host);
+
+  try
+  {
+    auto response = http::Client::send(request);
+    cout << response.getReasonPhrase();
+  }
+  catch (http::CouldNotResolveHostException& e)
+  {
+    cout << e.what() << endl;
+  }
+  catch (const char* msg)
```

Comment:
Yikes! We probably shouldn't be throwing `const char*`. Unless you have this here for temporary testing?

Reply:
Ok, I will create a custom exception for this as well XD.
Why don't we like this type of exceptions?